src/tests/test_engine.py

```diff
@@ -70,40 +71,41 @@ def setup_class(cls):
         gmxpath = which('mdrun_d') or which('gmx_d')
         gmxsuffix = '_d'
         # self.logger.debug("\nBuilding options for target...\n")
-        cls.cwd = os.path.dirname(os.path.realpath(__file__))
-        os.chdir(os.path.join(cls.cwd, "files", "amber_alaglu"))
-        if not os.path.exists("temp"):
-            os.makedirs("temp")
-        os.chdir("temp")
+        self.cwd = os.path.dirname(os.path.realpath(__file__))
+        os.chdir(os.path.join(self.cwd, "files", "amber_alaglu"))
+        tmpfolder = os.path.join(self.cwd, "files", "amber_alaglu", "temp")
+        if not os.path.exists(tmpfolder):
+            os.makedirs(tmpfolder)
+        os.chdir(tmpfolder)
         for i in ["topol.top", "shot.mdp", "a99sb.xml", "a99sb.prm", "all.gro", "all.arc", "AceGluNme.itp", "AceAlaNme.itp", "a99sb.itp"]:
             os.system("ln -fs ../%s" % i)
-        cls.engines = OrderedDict()
+        self.engines = OrderedDict()
         # Set up GMX engine
         if gmxpath != '':
-            cls.engines['GMX'] = GMX(coords="all.gro", gmx_top="topol.top", gmx_mdp="shot.mdp", gmxpath=gmxpath, gmxsuffix=gmxsuffix)
+            self.engines['GMX'] = GMX(coords="all.gro", gmx_top="topol.top", gmx_mdp="shot.mdp", gmxpath=gmxpath, gmxsuffix=gmxsuffix)
         else:
             logger.warn("GROMACS cannot be found, skipping GMX tests.")
         # Set up TINKER engine
         if tinkerpath != '':
-            cls.engines['TINKER'] = TINKER(coords="all.arc", tinker_key="alaglu.key", tinkerpath=tinkerpath)
+            self.engines['TINKER'] = TINKER(coords="all.arc", tinker_key="alaglu.key", tinkerpath=tinkerpath)
         else:
             logger.warn("TINKER cannot be found, skipping TINKER tests.")
         # Set up OpenMM engine
         try:
             import simtk.openmm
-            cls.engines['OpenMM'] = OpenMM(coords="all.gro", pdb="conf.pdb", ffxml="a99sb.xml", platname="Reference", precision="double")
+            self.engines['OpenMM'] = OpenMM(coords="all.gro", pdb="conf.pdb", ffxml="a99sb.xml", platname="Reference", precision="double")
         except:
             logger.warn("OpenMM cannot be imported, skipping OpenMM tests.")
         # the cleanup has been replaced by self.teardown_class
         # self.addCleanup(os.system, 'cd .. ; rm -rf temp')
 
-    @classmethod
-    def teardown_class(cls):
+    def teardown_method(self):
+
         """
         teardown any state that was previously setup with a call to setup_class.
         """
-        os.chdir('..')
-        shutil.rmtree('temp')
+        os.chdir(self.cwd)
+        shutil.rmtree(self.cwd, "files", "amber_alaglu", "temp")
```

Comment:
For now, let's keep test output (so don't remove files). If this causes subsequent tests to fail, we can reintroduce it. And later we can see about implementing cleanup conditional on tests passing.